frontend/src/data/providers/GunViolenceProvider.ts

```diff
@@ -0,0 +1,107 @@
+import { getDataManager } from '../../utils/globals'
+import { type DatasetId } from '../config/DatasetMetadata'
+import { type DataTypeId, type MetricId } from '../config/MetricConfig'
+import { type TimeView, type Breakdowns } from '../query/Breakdowns'
+import { type MetricQuery, MetricQueryResponse } from '../query/MetricQuery'
+import { appendFipsIfNeeded } from '../utils/datasetutils'
+import VariableProvider from './VariableProvider'
+
+export const GUN_VIOLENCE_DATATYPES: DataTypeId[] = [
+  'gun_violence',
+  'gun_violence_injuries',
+  'gun_violence_homicide',
+  'gun_violence_suicide',
+  'gun_violence_legal_intervention',
+]
+export const GUN_VIOLENCE_METRICS: MetricId[] = [
+  'gun_violence_homicide_per_100k',
+  'gun_violence_legal_intervention_per_100k',
+  'gun_violence_suicide_per_100k',
+  'gun_violence_injuries_per_100k',
+  'gun_violence_homicide_pct_share',
+  'gun_violence_legal_intervention_pct_share',
+  'gun_violence_suicide_pct_share',
+  'gun_violence_injuries_pct_share',
+  'gun_violence_homicide_pct_relative_inequity',
+  'gun_violence_legal_intervention_pct_relative_inequity',
+  'gun_violence_suicide_pct_relative_inequity',
+  'gun_violence_injuries_pct_relative_inequity',
+]
+const GUN_VIOLENCE_DETERMINANTS = [...GUN_VIOLENCE_METRICS]
+
+class GunViolenceProvider extends VariableProvider {
+  constructor() {
+    super('gun_violence_provider', GUN_VIOLENCE_DETERMINANTS)
+  }
+
+  getDatasetId(
```

Comment:
something in the arguments sent in to this method are not triggering any of the `if` conditions so it's returning `undefined`